rfc/0075-faster-failover-and-configuration-push.md

```diff
@@ -0,0 +1,315 @@
+# Meta
+
+| Field          | Value                                  |
+|----------------|----------------------------------------|
+| RFC Name       | Faster Failover and Configuration Push |
+| RFC ID         | 75                                     |
+| Start Date     | 2023-06-14                             |
+| Owner          | Sergey Avseyev                         |
+| Current Status | DRAFT                                  |
+| Revision       | #1                                     |
+
+# Summary
+
+TBD
+
+# Motivation
+
+TBD
+
+# Relation to Other RFCs
+
+This RFC relates to the following documents:
+
+* [RFC-0005][rfc-0005]: VBucket Retry Logic.
+
+* [RFC-0024][rfc-0024]: Fast-Failover SDK.
+
+
+# High-Level Design
+
+TBD
+
+# User-Facing API
+
+TBD
+
+# Implementation Details
+
+## Protocol Changes
+
+[https://issues.couchbase.com/browse/MB-57311]: #
+
+### Get Cluster Config with Known Version
+
+[https://review.couchbase.org/c/kv_engine/+/192301]: #
+
+The KV engine introduces a new HELLO flag called `GetClusterConfigWithKnownVersion` with a value of `0x1d`. This flag
+does not change the behavior of the server but allows determining if the node supports epoch-revision fields for the
+`GetClusterConfig` (`0xb5`) operation. If the node acknowledges `GetClusterConfigWithKnownVersion`, then the SDK can use
```

Comment:
If this is done on a per connection basis, rather than gated by a bucket capability to signal the whole cluster supports it, then won't this add quite a lot of complexity to SDKs? Especially in the mixer cluster case.

Reply:
I don't think so, because each connection knows its capability at the time the operation is being scheduled.

In fact you can always prepare `GetClusterConfigWithKnownVersion` (i.e. buffer where bodylen is set to `sizeof(uint64_t) * 2`, and fill the version, but when you schedule it to the connection that does not support new feature, you can set body length to zero.